### ISSUE TITLE
Refactor Feature class inheritance and types of id and properties members

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#39]
The `geocore` package (and also `geodata`) has dependency to the `attributes` package via following imports:

```dart
import 'package:attributes/entity.dart';
import 'package:attributes/values.dart';
```

It's used only for following things:
- `Feature` extends `Entity` from the `attributes` package
- `id` member of `Entity` is of the type `Identifider`
- `properties` member of `Entity` is of the type `DataObject`

References are located in following source code:
- `lib/src/feature/feature.dart`
- `lib/src/parse/geojson/geojson.dart`

Consider removing the dependency to the `attributes` package and use some more generic types instead.

Properties could be handled as `Map<String, Object?>` and id stored as String (even if both number and text representations could occur for feature ids in GeoJSON for example).

This change would allow more generic usage of the `geocore` library and `Feature` classes it defines without introducing a dependency to a single package (that is now used only very partially). 

Any side effects from this change?

How to check also the `geodata` package.

**Comment by navispatial** [comment]
Before the change the abstract `Feature` class contained following public members, two first via extending `Entity`:
```dart
  /// An optional [id] for this entity.
  Identifier? get id;

  /// The required [properties] for this entity, allowed to be empty.
  DataObject get properties;

  /// An optional [geometry] for this feature.
  Geometry? get geometry;
```

Here [Entity](https://pub.dev/documentation/attributes/0.8.0/entity/Entity-class.html), [Identifier](https://pub.dev/documentation/attributes/0.8.0/values/Identifier-class.html) and [DataObject](https://pub.dev/documentation/attributes/0.8.0/data/DataObject-class.html) are defined by the separate [attributes](https://pub.dev/packages/attributes) package.

After the change the abstract `Feature` class contains following public members without need to extends `Entity`:
```dart
  /// An optional identifier for this feature.
  ///
  /// Note that an identifier could be textual or a number but reprensented here
  /// as a nullable String object.
  String? get id;

  /// Required properties for this feature allowed to be empty.
  Map<String, Object?> get properties;

  /// An optional geometry for this feature.
  Geometry? get geometry;
```

So dependency to the [attributes](https://pub.dev/packages/attributes) package is removed. The `Feature` class still has [Geometry](https://pub.dev/documentation/geocore/latest/base/Geometry-class.html) but it's defined by the *geocore* itself.

Still if properties from a feature object should be used via [DataObject](https://pub.dev/documentation/attributes/0.8.0/data/DataObject-class.html) it's possible to
* access properties as the standard `Map<String, Object?>` data from an feature
* wrap data with `DataObject` and use accessors defined by it
* see sample below

```dart
import `package:geocore/feature.dart`;
import `package:attributes/data.dart`;

// get a feature object some where
final Feature myFeature = getFeature();

// wrap properties with DataObject
final myProperties = DataObject.view(myFeature.properties);

// use accessors
final name = myProperties.getString('name');
final measure = myProperties.getDouble('measure');
final subObject = myProperties.object('sub');
final subName = subObject.getString('name');
```

**Comment by navispatial** [comment]
[Implemented on the pre-release 0.8.0-a.2 of the upcoming 0.8.0 release](https://github.com/navibyte/geospatial/blob/main/dart/geocore/CHANGELOG.md#080)
